docs/source/tutorial/performance.rst

```diff
@@ -150,3 +150,7 @@ automatic memory optimization determines the maximum sub-batch size for training
 above described process :ref:`sub_batching`. The batch sizes are determined using binary search taking into
 consideration the `CUDA architecture <https://developer.download.nvidia.com/video/gputechconf/gtc/2019/presentation/s9926-tensor-core-performance-the-ultimate-guide.pdf>`_
 which ensures that the chosen batch size is the most CUDA efficient one.
+
+Fallback to Evaluation on CPU
+-----------------------------
```

Comment:
@lvermue any chance you'd be willing to rewrite some of the discussion from this PR for users who might want to understand the context of this PR?